loader_hub/remote_depth/base.py

```diff
@@ -0,0 +1,90 @@
+"""Remote file reader.
+
+A loader that fetches an arbitrary remote page or file by URL and parses its contents.
+
+"""
+from typing import Any, Dict, List, Optional, Union
+from urllib.parse import urljoin, urlparse, urlunparse
+
+import requests
+from bs4 import BeautifulSoup
+from llama_index import download_loader
+from llama_index.readers.base import BaseReader
+from llama_index.readers.schema.base import Document
+from tqdm.auto import tqdm
+
+
+class RemoteDepthReader(BaseReader):
```

Comment:
I think it's ok to have this as a separate class for now. IMO it's closer to a web scraper than `RemoteReader`.

Reply:
Yes, correct, Should we rename that to `WebScraperReader` when we have more functionalities, and keep that named `RemoteDepthReader` for now?

Reply:
Yeah let's keep it for now. There are other web scrapers in the library already. We can reorganize down the road if needed.